drivers/watchdog/wdt_cc13xx_cc26xx.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2022 Florin Stancu <niflostancu@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_cc13xx_cc26xx_watchdog
+
+#include <zephyr/drivers/watchdog.h>
+#include <soc.h>
+#include <errno.h>
+
+#define LOG_LEVEL CONFIG_WDT_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(wdt_cc13xx_cc26xx);
+
+/* Driverlib includes */
+#include <inc/hw_types.h>
+#include <inc/hw_wdt.h>
+#include <driverlib/rom.h>
+#include <driverlib/prcm.h>
+#include <driverlib/watchdog.h>
+
+#define CPU_FREQ DT_PROP(DT_PATH(cpus, cpu_0), clock_frequency)
+
+#define WDT_INST_IDX            0
```

Comment:
?

Reply:
I needed the system clock to convert milliseconds to WDT reload ticks.
I don't remember where I got this, most probably from here: [`log_backend_swo.c`](https://github.com/zephyrproject-rtos/zephyr/blob/f249f278bf586c87b89b7360d65977d71a236f21/subsys/logging/log_backend_swo.c#L49).
Is there another / preferred way to read the system clock?

As for `WDT_INST_IDX`, I copied some driver initialization code which used `DT_INST_FOREACH_*` and I renamed the index parameter to this.
Should I expand the constant `0` everywhere or should I rewrite it using `DT_INST_FOREACH_*` regardless of the WDT being a singleton? (defined code is pretty hard to read :D)

Thanks.

Reply:
Yes, please use `DT_INST_FOREACH_*()`

Reply:
Okay, I'll get to it tomorrow ;)